build/smartmontools/files/system-smartd.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0"?>
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<!--
+CDDL HEADER START
+
+The contents of this file are subject to the terms of the
+Common Development and Distribution License (the "License").
+You may not use this file except in compliance with the License.
+
+You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+or http://www.opensolaris.org/os/licensing.
+See the License for the specific language governing permissions
+and limitations under the License.
+
+When distributing Covered Code, include this CDDL HEADER in each
+file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+If applicable, add the following below this CDDL HEADER, with the
+fields enclosed by brackets "[]" replaced with your own identifying
+information: Portions Copyright [yyyy] [name of copyright owner]
+
+CDDL HEADER END
+-->
+
+<!--
+    Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
+    Use is subject to license terms.
+-->
+<service_bundle type="manifest" name="smartd">
+<service name="system/smartd" type="service" version="1">
+    <create_default_instance
+        enabled="false" />
+
+    <single_instance/>
+
+    <dependency name="filesystem-local" grouping="require_all" restart_on="none" type="service">
+        <service_fmri value="svc:/system/filesystem/local:default"/>
+    </dependency>
+    <dependency name="autofs" grouping="optional_all" restart_on="error" type="service">
+        <service_fmri value="svc:/system/filesystem/autofs:default"/>
+    </dependency>
+    <dependency name="config-file" grouping="require_all" restart_on="refresh" type="path">
+        <service_fmri value="file://localhost/etc/opt/ooce/smartmontools/smartd.conf" />
+    </dependency>
+
+    <exec_method
+        type="method"
+        name="start"
+        exec="/opt/ooce/sbin/smartd --pidfile=/var/opt/ooce/smartmontools/run/smartd.pid --savestates=/var/opt/ooce/smartmontools/"
```

Comment:
there is a /var/run directory, mounted to swap, so that it gets flushed on reboot ... I guess it would make sense to have our pid files there too ... `/var/run/opt/ooce/...`  as this avoids stale pid files after reboot.

Reply:
well we'd need to add wrapper scripts to create the subdirectories in `/var/run` if they don't exist (i.e. after every reboot).